src/software/amazon/ion/impl/lite/IonIntLite.java

```diff
@@ -34,11 +35,18 @@
     static private final BigInteger LONG_MAX_VALUE =
         BigInteger.valueOf(Long.MAX_VALUE);
 
+    static private final BigInteger INT_MIN_VALUE =
+        BigInteger.valueOf(Integer.MIN_VALUE);
+
+    static private final BigInteger INT_MAX_VALUE =
+        BigInteger.valueOf(Integer.MAX_VALUE);
+
     private static final int HASH_SIGNATURE =
         IonType.INT.toString().hashCode();
 
     private long _long_value;
     private BigInteger _big_int_value;
+    private IntegerType _actual_type;
```

Comment:
Is it really worth spending another pointer per node on this?  We're now spending two pointers and a `long` to redundantly-represent data that _could_ be done with only a `BigInteger`. So I bet there's a more compact representation.